### ISSUE TITLE
GPU version

### ISSUE THREAD

**Issue by SuperWu090** [wentianli/knowledge_distillation_caffe#6]
Hi， 
    I am very intersted in this repo. I wonder if you have tried to implemented the GPU version ?
    Thanks very much !

**Comment by wentianli** [comment]
@SuperWu090 In my implementation, the softmax part is computed using two built-in softmax layers which can run on GPU, while other parts (addition, multiplication, log, etc) run on CPU. It's not very slow, so I don't plan to write a .cu file.

**Comment by jianbaba** [comment]
Hi，Thank you for your great work.
when,I run the training,I have a question:
I0920 11:00:58.925289  7154 caffe.cpp:248] Starting Optimization
I0920 11:00:58.925359  7154 solver.cpp:272] Solving SpherefaceNet-64
I0920 11:00:58.925410  7154 solver.cpp:273] Learning Rate Policy: multistep
*** Aborted at 1568948459 (unix time) try "date -d @1568948459" if you are using GNU date ***
PC: @     0x7f23624fd8f5 caffe::Blob<>::mutable_cpu_data()
*** SIGSEGV (@0x0) received by PID 7154 (TID 0x7f23634da740) from PID 0; stack trace: ***
    @     0x7f234b109370 (unknown)
    @     0x7f23624fd8f5 caffe::Blob<>::mutable_cpu_data()
    @     0x7f236251c49a caffe::MarginInnerProductLayer<>::Forward_gpu()
    @     0x7f23624dc0f3 caffe::Net<>::ForwardFromTo()
    @     0x7f23624dc4b7 caffe::Net<>::Forward()
    @     0x7f23624d5138 caffe::Solver<>::Step()
    @     0x7f23624d5aea caffe::Solver<>::Solve()
    @           0x40b69a train()
    @           0x408a0c main
    @     0x7f234ad5ab35 __libc_start_main
    @           0x40937d (unknown)

Is it because, I am using the gpu version of caffe, and here only provides the cp version of the kl layer？

**Comment by wentianli** [comment]
@jianbaba I don't think the layer is the cause.  The gpu layer is not obligatory for the gpu version of caffe. Maybe you need to check whether the hyperparameters for 'multistep' learning rate policy are set properly.
